src/web/app/components/question-types/question-additional-info/mcq-question-additional-info.component.spec.ts

```diff
@@ -22,4 +27,83 @@ describe('McqQuestionAdditionalInfoComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default view', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  const mcqDetail: FeedbackMcqQuestionDetails = {
+    hasAssignedWeights: false,
+    mcqWeights: [],
+    mcqOtherWeight: 0,
+    numOfMcqChoices: 3,
+    mcqChoices: ['1', '2', '3'],
+    otherEnabled: false,
+    generateOptionsFor: FeedbackParticipantType.NONE,
+    questionType: FeedbackQuestionType.MCQ,
+    questionText: 'MCQ question detail',
+  };
+
+  const mcqDetailWithOther: FeedbackMcqQuestionDetails = {
+    hasAssignedWeights: false,
+    mcqWeights: [],
+    mcqOtherWeight: 0,
+    numOfMcqChoices: 3,
+    mcqChoices: ['1', '2', '3'],
+    otherEnabled: true,
+    generateOptionsFor: FeedbackParticipantType.NONE,
+    questionType: FeedbackQuestionType.MCQ,
+    questionText: 'MCQ question detail',
+  };
+
+  const mcqDetailWithGenerateOption: FeedbackMcqQuestionDetails = {
+    hasAssignedWeights: false,
+    mcqWeights: [],
+    mcqOtherWeight: 0,
+    numOfMcqChoices: 3,
+    mcqChoices: ['1', '2', '3'],
+    otherEnabled: false,
+    generateOptionsFor: FeedbackParticipantType.STUDENTS,
+    questionType: FeedbackQuestionType.MCQ,
+    questionText: 'MCQ with generate options question detail',
+  };
+
+  it('should show mcq options', () => {
+    component.questionDetails = mcqDetail;
+    fixture.detectChanges();
+    const generateOptionFor: HTMLElement = fixture.nativeElement.querySelector('div');
+    const choices: HTMLElement[] = fixture.nativeElement.querySelectorAll('li');
+    expect(generateOptionFor).toBeNull();
+    expect(choices.length).toEqual(3);
+    expect(choices[0].textContent).toEqual('1');
+    expect(choices[1].textContent).toEqual('2');
+    expect(choices[2].textContent).toEqual('3');
+    expect(fixture).toMatchSnapshot();
```

Comment:
Will snapshot cover all the verification above?
```typescript
    expect(generateOptionFor).toBeNull();
    expect(choices.length).toEqual(3);
    expect(choices[0].textContent).toEqual('1');
    expect(choices[1].textContent).toEqual('2');
    expect(choices[2].textContent).toEqual('3');
```

If that is the case, you don't need them?